src/veil/ringct/lightwallet.cpp

```diff
@@ -0,0 +1,1890 @@
+    }
+
+    // Get Tx From API -
+    std::vector<std::pair<int, CWatchOnlyTx>> vTxes;
+    FetchWatchOnlyTransactions(scan_secret, vTxes);
+
+    //Build correct vector
+    std::vector<CWatchOnlyTx> myTxes;
+    for(const auto& pair: vTxes){
+        myTxes.push_back(pair.second);
+    }
+
+    // Get Keyimages
+    std::vector<std::pair<CCmpPubKey, CWatchOnlyTx>> keyimages;
+    if (!GetKeyImagesFromAPITransactions(keyimages, myTxes, spend_secret, scan_secret, spend_pubkey, errorMsg)) {
+        LogPrintf("Failed - %s\n", errorMsg);
+        return errorMsg;
+    }
+
+    // Check if key images are spent
+    std::vector<CWatchOnlyTx> vUnspentTxes;
+    for (const auto& pair : keyimages) {
+        uint256 tx_hash;
+        bool spent_in_chain = pblocktree->ReadRCTKeyImage(pair.first, tx_hash);
+        if (!spent_in_chain) {
+            vUnspentTxes.push_back(pair.second);
+        }
+    }
+
+    // Get Amount and blinds of all spendable txes
+    std::vector<CWatchOnlyTx> vFinalTx;
+    if (!GetAmountAndBlindForUnspentTx(vFinalTx, vUnspentTxes, spend_secret, scan_secret, spend_pubkey,errorMsg)) {
+        LogPrintf("Failed - %s\n", errorMsg);
+        return errorMsg;
+    }
+
+    // Get Random outputs
+    std::set<int64_t> setIndexSelected;
+    std::vector<CLightWalletAnonOutputData> vecSelectedOutputs;
+    std::string sError;
+    GetMainWallet()->GetAnonWallet()->GetRandomHidingOutputs(11, 11, setIndexSelected,vecSelectedOutputs, errorMsg);
+
+    std::string txHex = "";
+
+    if (!BuildLightWalletTransaction(args, vFinalTx, vecSelectedOutputs, txHex, errorMsg)) {
+        LogPrintf("Failed testing transaction");
+    }
+
+    return txHex;
+}
+
+
+bool BuildLightWalletTransaction(const std::vector<std::string>& args, const std::vector<CWatchOnlyTx>& vSpendableTx, const std::vector<CLightWalletAnonOutputData>& vDummyOutputs, std::string& txHex, std::string& errorMsg)
+{
+
+    // We need to direct this to spend the correct outputs.
+    // If the vSpendableTx has Anon and Stealth outputs we will use Anon
+
+    int nAnonTxes = 0;
+    int nStealthTxes = 0;
+
+    std::vector<CWatchOnlyTx> vAnonTxes;
+    std::vector<CWatchOnlyTx> vStealthTxes;
+
+
+    for (const auto& tx : vSpendableTx) {
+        if (tx.type == CWatchOnlyTx::ANON) {
+            vAnonTxes.push_back(tx);
+            nAnonTxes++;
+        }
+
+        if (tx.type == CWatchOnlyTx::STEALTH) {
+            vStealthTxes.push_back(tx);
+            nStealthTxes++;
+        }
+    }
+
+
+    if (nAnonTxes > 0) {
+        // We use Anon / RingCT
+        return BuildLightWalletRingCTTransaction(args, vAnonTxes, vDummyOutputs, txHex, errorMsg);
+    } else if (nStealthTxes > 0) {
+        return BuildLightWalletStealthTransaction(args, vStealthTxes, txHex, errorMsg);
+    } else {
+        errorMsg = "No Anon or Stealth txes given to build transaction";
+        return false;
+    }
+
+
+}
+
+
+/** Before you build the transaction, you must of already called
```

Comment:
It should be reasonable to perform these safety checks here anyway. And, as I mentioned in review comments on the docs, sometimes it's better to perform that work here instead of expecting it to be done.